### ISSUE TITLE
E-Distance: Choice whether to square Euclidean Distance

### ISSUE THREAD

**Issue by eroell** [theislab/pertpy#388]
### Report

Hi,

a short note on the E-distance:
The documentation of [pertpy.tools.Distance](https://pertpy.readthedocs.io/en/latest/usage/tools/pertpy.tools.Distance.html#pertpy.tools.Distance) at the moment refers to the [preprint of scPerturb](https://doi.org/10.1101/2022.08.20.504663), where the E-distance is nicely described.

In the preprint, the E-distance is described as "the squared euclidean distance when calculating cell-wise distances".

The `sklearn.metrics.pairwise_distances` used for the [E-distance in pertpy](https://github.com/theislab/pertpy/blob/54f34d15efa34b6b976763d54dc5326cd63a81db/pertpy/tools/_distances/_distances.py#L308) computes the Euclidean distance (so "without squaring", or when considering the formulas on p.21 and p.22 in the preprint, technically adding a root in these formulas).

While not changing the order of distances, the lacking squaring of distances might be unexpected.

Apologies if I confused something here!

**Comment by stefanpeidli** [comment]
Hi Eljas,

Thanks for the note! I changed the cell-wise distance for Edistance to squared euclidean now. I needed to change the way we handle stuff internally for this (me dreading to do this was the main reason it was "euclidean" until now, but I think it worked out fine).

I've pushed the changes to the implement_additional_distance_metrics branch for now (see https://github.com/theislab/pertpy/commit/c5c2be04c80ec5265e6ebcf19de82949ec1f50ce).
@Zethson are you taking care of pulling the changes to main or should I?
